parser/parser.go

```diff
@@ -133,6 +140,29 @@ func gSelect() (*NodeSelect, error) {
 	return s, nil
 }
 
+func gShow() (*NodeShow, error) {
+	var err *lexical.TokenError
+	look_ahead, err = lexical.Token()
+	if err != nil {
+		return nil, err
+	}
+
+	node := new(NodeShow)
+	switch look_ahead {
+	case lexical.T_TABLES:
+		node.Tables = true
+		break
+	case lexical.T_DATABASES:
+		node.Databases = true
+		break
+	default:
+		fmt.Println("huh", look_ahead)
+		return nil, fmt.Errorf("Error: can only show tables or databases")
+	}
+	look_ahead, _ = lexical.Token()
```

Comment:
I think we need to add some unit tests to make sure that it's not possible to run `show tables show` for example. This line seems allowing any token after `show tables` and it should be just EOF.

Reply:
addedd tests in 22d91b6